x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/ReferenceAttribute.java

```diff
@@ -117,4 +117,9 @@ protected NodeInfo<ReferenceAttribute> info() {
     protected String label() {
         return "r";
     }
+
+    @Override
+    public boolean isDimension() {
+        return false;
```

Comment:
thought: it might get tricky when dealing with renames; if you have a `FROM idx | RENAME dimension_field AS foo`, then the reference attribute for `foo` will claim not to be a dimension - so downstream plan nodes may need to resolve aliases to correctly determine if they're looking at a dimension or not.

Reply:
My understanding from @kkrik-es is that, at least for the initial version, that's an acceptable restriction.  I think ReferenceAttributes are used for the output of EVAL and similar commands, and at least for the initial version of this we do not want to treat those as dimensions.